odbc/src/api/bind_col_tests.rs

```diff
@@ -466,4 +403,832 @@ mod unit {
+                    doc! {"test": {"num": 40, "word": "dddd"}},
+                    doc! {"test": {"num": 50, "word": "eeee"}},
+                    doc! {"test": {"num": 60, "word": "ffff"}},
+                    doc! {"test": {"num": 70, "word": "gggg"}},
+                    doc! {"test": {"num": 80, "word": "hhhh"}},
+                ],
+                vec![
+                    MongoColMetadata::new(
+                        "",
+                        "test".to_string(),
+                        "num".to_string(),
+                        Schema::Atomic(Atomic::Scalar(BsonTypeName::Int)),
+                        Nullability::SQL_NO_NULLS,
+                        TypeMode::Simple,
+                    ),
+                    MongoColMetadata::new(
+                        "",
+                        "test".to_string(),
+                        "word".to_string(),
+                        Schema::Atomic(Atomic::Scalar(BsonTypeName::String)),
+                        Nullability::SQL_NO_NULLS,
+                        TypeMode::Simple,
+                    ),
+                ],
+            );
+
+            // Set the mongo_statement
+            *s.mongo_statement.write().unwrap() = Some(Box::new(mock_query));
+
+            // assert that SQLFetch is successful. We are fetching the first 4 rows in the result set.
+            assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt as *mut _));
+
+            // assert that the first 4 values from the result set were put in the bound buffer array correctly for column 1
+            let bound_buffer = s
+                .bound_cols
+                .read()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get(&1)
+                .unwrap()
+                .target_buffer;
+            assert_eq!(10, *(bound_buffer as *mut i32));
+            assert_eq!(20, *((bound_buffer as ULen + 4) as *mut i32));
+            assert_eq!(30, *((bound_buffer as ULen + 8) as *mut i32));
+            assert_eq!(40, *((bound_buffer as ULen + 12) as *mut i32));
+
+            // assert that the indicator has the correct values for column 1.
+            let indicator = s
+                .bound_cols
+                .read()
+                .unwrap()
+                .as_ref()
+                .unwrap()
+                .get(&1)
+                .unwrap()
+                .length_or_indicator;
+            assert_eq!(4, *indicator);
+            assert_eq!(4, *((indicator as ULen + 8) as *mut Len));
+            assert_eq!(4, *((indicator as ULen + 16) as *mut Len));
+            assert_eq!(4, *((indicator as ULen + 24) as *mut Len));
+
+            // assert that SQLFetch is successful. We are fetching the next 4 rows in the result set.
+            assert_eq!(SqlReturn::SUCCESS, SQLFetch(stmt as *mut _));
+
+            // assert that the last 4 values from the result set were put in the bound buffer array correctly for column 1
+            let bound_buffer = s
```

Comment:
I can actually remove all the "getting" I do in all these tests if we would prefer it that way. The "getting" isn't actually necessary for any of the tests since I create all the references at the top of the tests. I just did that to check everything in the process; however, since the likelihood of this part actually catching an issue is so small, I can just remove it.